lib/Pusher.php

```diff
@@ -20,9 +20,9 @@
 		+ Phil Leggetter (phil@leggetter.co.uk)
 */
 
-class PusherException extends Exception
-{
-}
+require_once('PusherException.php');
+require_once('PusherHTTPException.php');
+require_once('TriggerResult.php');
```

Comment:
And you have 2 solutions for that:

- using the classmap autoloading for Composer (which you already btw)
- split each class into a separate file (named after the class) and rely on PSR-0 autoloading for them

For composer users, this will be equivalent (given that all the work will be done by Composer anyway). There are 2 cases where it would make a difference:

- for contributors of the library: when using classmap autoloading, it requires regenerating the classmap when adding a new class, while PSR-0 can find it as soon as the file exists in the right place
- in case you want to ship your own autoloader in the library for easier consumption by people not using Composer, relying on a classmap would force having the classmap in the repo (so thinking about updating it when adding a new class) while PSR-0 would not require any new change to the autoloader class once implemented.

If you tell me what you prefer, I can even send you a PR with the change